src/generator.rs

```diff
@@ -150,16 +151,24 @@ fn write_contents(output_directory: &Path, filename: &str, contents: &str) -> Re
     let path = output_directory.join(filename);
     make_parent(&path)?;
 
-    let contents = format!(
-        "\
-# Automatically generated by {exe_name}
-
-{contents}",
-        exe_name = std::env::current_exe().unwrap().display(),
-        contents = contents
-    );
-
-    fs::write(&path, contents).with_context(|| format!("Failed to write {}", path.display()))
+    // TODO: ideally this'd be write_all_vectored and we'd both Not Allocate
+    //       /and/ complete in one syscall, but that's an unstable feature
+    fs::File::create(&path)
+        .map(io::BufWriter::new)
+        .and_then(|mut f| {
+            f.write_all(b"# Automatically generated by ")?;
+            f.write_all(
+                env::current_exe()
+                    .as_deref()
+                    .unwrap_or_else(|_| Path::new(concat!(crate_name!(), " ", crate_version!())))
+                    .as_os_str()
+                    .as_bytes(),
+            )?;
+            f.write_all(b"\n\n")?;
+            f.write_all(contents.as_bytes())?;
+            f.flush()
+        })
+        .with_context(|| format!("Failed to write {}", path.display()))
```

Comment:
I don't find this to be an improvement… The code is longer and **much** harder to read. If this was in some hot path, then sure, it'd make sense. But we generally create a few files, and even if we "overallocate" here, this doesn't matter at all.